managed_config/index.bs

```diff
@@ -0,0 +1,98 @@
+
+## Managed configuration ## {#managed-configuration}
+
+For the purposes of this specification the device administrator can control a list of managed web applications. Each entry in this list can have a set JSON configuration, which is accessible by the documents hosted at the origin of the web application.
+
+Note that this does not apply to anonymous contexts, where the observed state is indistinguishable from a non-managed one.
+
+## <dfn dfn>Data model</dfn> ## {#data-model-section}
+
+We can assume without loss of generality that managed configuration is stored in a two-level key-value store. The first level of this store corresponds to configured origins, while the second corresponds to per-origin configuration string keys.
+## Data integrity verification ## {#data-integrity-verification}
+
+There is no managed configuration verification mechanism defined in the specification, which implies that the website should assume that the configuration could be tampered with, stolen or be replayed on a non-managed device. 
+
+Alternative security measures should be put in place by the website itself.
+
+# Extensions to the {{Navigator}} interface # {#navigator-extensions}
+
+<xmp class="idl">
+[
+  SecureContext
+] partial interface Navigator {
+  [SecureContext, SameObject]
+  readonly attribute NavigatorManagedData managed;
+};
+</xmp>
+
+
+{{managed}} attribute
+When getting, the managed attribute always returns the same instance of the {{NavigatorManagedData}} object.
+
+# {{NavigatorManagedData}} interface # {#navigatormanageddata-interface}
+
+<xmp class="idl">
+[
+  SecureContext,
+  Exposed=Window
+] interface NavigatorManagedData : EventTarget {
+  // Managed Configuration API.
+  Promise<object> getManagedConfiguration(sequence<DOMString> keys);
+  attribute EventHandler onmanagedconfigurationchange;
+};
+</xmp>
+
+## {{getManagedConfiguration()}} method ## {#getmanagedconfiguration-method}
+
+The {{getManagedConfiguration()}} method steps are:
+1. Let |promise| be a new promise.
+1. Run the following steps in parallel:
+    1. Try requesting managed configuration by sending {{keys}} to the [=data model=].
+    1. If the application is not managed, reject |promise| with a {{NotAllowedError}} DOMException.
+    1. Create an empty javascript |object|.
+    1. For each of the keys that had an associated  configuration in the [=data model=], add an attribute to the |object|.
+    1. Resolve |promise| with |object|.
+1. Return |promise|.
+
+## {{onmanagedconfigurationchange}} attribute ## {#onmanagedconfigurationchange-attribute}
+
+{{onmanagedconfigurationchange}} is an event handler IDL attribute for the |managedconfigurationchange| event type.
+
+When any of the configuration values under the origin-level key changes for the current origin, |managedconfigurationchange| event is fired with no value associated with it.
```

Comment:
We can specify these steps a little more precisely by saying that the user agent should iterate over each instance of NavigatorManagedData and queue a task to fire an event at it.

Reply:
Chatting with @domenic we think the proper incantation is, "For each instance data of NavigatorManagedData: let document be data's relevant global object's associated Document. If document is fully active and document's origin is ...: